### PR TITLE
Fix include grouping

### LINKED ISSUE #324: Add includes grouping to .clang-format
Included headers should be grouped together according to Google C++ Style Guide rules: https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes

This can be added to .clang-format. See "IncludeCategories" at https://clang.llvm.org/docs/ClangFormatStyleOptions.html.

### PR DESCRIPTION
**Pull request by mkow**
Regroups includes to conform to Google style. Also fixes some unnecessary/doubled imports.

Fixes #324 and #345.

### CODE REVIEW

File: src/parser/parser.cc [hunk 1]
```diff
@@ -16,6 +16,7 @@
  */
 
 #include "parser/parser.h"
+
```
Comment (catsuryuu) [inline src/parser/parser.cc]: Why empty line here?

File: src/ui/connectiondialog.cc [hunk 2]
```diff
@@ -28,6 +27,7 @@
 #include <QtGlobal>
 
 #include "client/networkclient.h"
+#include "ui_connectiondialog.h"
```
Comment (catsuryuu) [inline src/ui/connectiondialog.cc]: What is it?

File: src/ui/createchunkdialog.cc [hunk 3]
```diff
@@ -14,7 +14,8 @@
  * limitations under the License.
  *
  */
-#include "include/ui/createchunkdialog.h"
+#include "ui/createchunkdialog.h"
+
 #include "ui_createchunkdialog.h"
```
Comment (catsuryuu) [inline src/ui/createchunkdialog.cc]: What is it? (2)

File: src/visualization/samplingmethoddialog.cc [hunk 4]
```diff
@@ -14,12 +14,12 @@
...
  */
+#include <limits>
+
 #include <QComboBox>
 
-#include "include/visualization/samplingmethoddialog.h"
 #include "ui_samplingmethoddialog.h"
-
-#include <limits>
+#include "visualization/samplingmethoddialog.h"
```
Comment (catsuryuu) [inline src/visualization/samplingmethoddialog.cc]: Should be on top.

File: include/data/nodeid.h [hunk 5]
```diff
@@ -19,9 +19,10 @@
 #include <random>
 #include <vector>
 
-#include <QString>
 #include <msgpack.hpp>
```
[the remark this answers is not in the document]
Comment (mkow) [reply include/data/nodeid.h]: Done.

File: src/visualization/samplingmethoddialog.cc (hunk 4 above)
Comment (mkow) [reply src/visualization/samplingmethoddialog.cc]: Fixed.

**Pull request merged**